dids/diddht/resolver_test.go

```diff
@@ -0,0 +1,119 @@
+package diddht
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/alecthomas/assert/v2"
+)
+
+const dhtSpecVectors string = "../../web5-spec/test-vectors/did_dht/resolve.json"
+
+type vector struct {
+	Description string `json:"description"`
+	Input       struct {
+		DIDUri string `json:"didUri"`
+	} `json:"input"`
+	Output struct {
+		DIDResolutionMetadata struct {
+			Error string `json:"error"`
+		} `json:"didResolutionMetadata"`
+	} `json:"output"`
+	Errors bool `json:"errors"`
+}
+
+func initVector() ([]vector, error) {
+	// Load test vectors from file
+	data, err := os.ReadFile(dhtSpecVectors)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal test vectors
+	vectorData := struct {
+		Vectors []vector `json:"vectors"`
+	}{}
+	if err := json.Unmarshal(data, &vectorData); err != nil {
+		return nil, err
+	}
+	return vectorData.Vectors, nil
+}
+
+func Test_resolve_failures(t *testing.T) {
```

Comment:
Currently all the available test vectors are failure cases but eventually [that test vector](https://github.com/TBD54566975/web5-spec/blob/main/test-vectors/did_dht/resolve.json) should include many cases, only some of which are failure cases. For example, [web5-js has a single test case](https://github.com/TBD54566975/web5-js/blob/91d52aaa9410db5e5f7c3c31ebfe0d4956028496/packages/dids/tests/methods/did-dht.spec.ts#L1154).

```suggestion
func Test_VectorsResolve(t *testing.T) {
```